File: backend/app/schemas/sandbox.py
"""Sandbox API schemas."""

from typing import Any, Dict, List
from pydantic import BaseModel, Field

class SandboxAnalyzeRequest(BaseModel):
    """Request model for clean text analysis."""
    content: str = Field(..., min_length=10, description="Markdown content to analyze")

class PillarScoreSchema(BaseModel):
    """Schema for individual pillar scores."""
    score: float
    max_score: float
    reason: List[str]

class AEOScoreResultSchema(BaseModel):
    """Schema for AEO score results."""
    total_score: float
    rubric_version: str
    pillars: Dict[str, PillarScoreSchema]
    details: Dict[str, Any]

class SandboxAnalyzeResponse(BaseModel):
    """Response model for sandbox analysis."""
    aeo: AEOScoreResultSchema

# --- AI Sandbox Schemas ---

class SandboxAIRequest(BaseModel):
    content: str = Field(..., min_length=10, description="Content to check for AI likeness")

class SubScoreItem(BaseModel):
    score: float
    max_score: float
    percentage: float
    explanation: str
    evidence: List[str]

class RubricRawResponse(BaseModel):
    rubric_version: str
    total_score: float
    subscores: Dict[str, SubScoreItem]
    metadata: Dict[str, Any]

class SandboxAIResponse(BaseModel):
    """Response model for AI sandbox analysis."""
    score: float
    model_version: str
    timestamp: str
    raw_response: RubricRawResponse

File: frontend/lib/api/sandbox.ts
/**
 * AEO Sandbox API Client
 */

const API_URL = (process.env.NEXT_PUBLIC_API_URL || 'http://localhost:8000') + '/api/v1'

export interface PillarScore {
    score: number
    max_score: number
    reason: string[]
}

export interface AEOScoreResult {
    total_score: number
    rubric_version: string
    pillars: Record<string, PillarScore>
    details: any
}

export interface SandboxAnalyzeResponse {
    aeo: AEOScoreResult
}

// --- AI Sandbox Types ---

export interface SubScoreItem {
    score: number
    max_score: number
    percentage: number
    explanation: string
    evidence: string[]
}

export interface RubricRawResponse {
    rubric_version: string
    total_score: number
    subscores: Record<string, SubScoreItem>
    metadata: any
}

export interface SandboxAIResponse {
    score: number
    model_version: string
    timestamp: string
    raw_response: RubricRawResponse
}

export async function analyzeAEOSandbox(
    content: string,
    token: string
): Promise<SandboxAnalyzeResponse> {
    const response = await fetch(`${API_URL}/sandbox/aeo`, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            Authorization: `Bearer ${token}`,
        },
        body: JSON.stringify({ content }),
    })

    if (!response.ok) {
        const error = await response.json().catch(() => ({ detail: 'Unknown error' }))
        throw new Error(error.detail || `HTTP ${response.status}`)
    }

    return response.json()
}

export async function analyzeAISandbox(
    content: string,
    token: string
): Promise<SandboxAIResponse> {
    const response = await fetch(`${API_URL}/sandbox/ai`, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            Authorization: `Bearer ${token}`,
        },
        body: JSON.stringify({ content }),
    })

    if (!response.ok) {
        const error = await response.json().catch(() => ({ detail: 'Unknown error' }))
        throw new Error(error.detail || `HTTP ${response.status}`)
    }

    return response.json()
}
